### PR TITLE
fix(quantic): fix the facet when click on collapse/expand button

### PR DESCRIPTION
**Pull request by aelfaleh**
The bug happend when we insert text in the iput search of the facet, then we click collapse button and expand button => we got this error message 

image

I also add some other verification to avoid problem like we had this morning when the search results are null

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js [hunk 1]
```diff
@@ -140,14 +140,14 @@ export default class QuanticFacet extends LightningElement {
 
   get facetSearchResults() {
     const results = this.facet.state.facetSearch.values;
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:142]: `hasSearchResults` would also fail when `this.facet.state.facetSearch.values` is `null`.

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js [hunk 2]
```diff
@@ -127,15 +127,15 @@ export default class QuanticCategoryFacet extends LightningElement {
 
   get facetSearchResults() {
     const results = this.facet.state.facetSearch.values;
-    return results.map((result, index) => ({
+    return results?.map((result, index) => ({
```
[the remark this answers is not in the document]
Comment (aelfaleh) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js:130]: good catch

**Review by nathanlb — APPROVED** [review]
@lbergeron makes a good point that i think should be addressed but otherwise 👍

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js:130]: Are we sure `this.facet.state.facetSearch.values` no part of this is ever undefined? If I am not mistaken, JS still throws if any part is undefined when trying to access the value `facetSearch` of `undefined` for example, even with the `??` operator.

image

File: packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js [hunk 3]
```diff
@@ -140,14 +140,14 @@ export default class QuanticFacet extends LightningElement {
...
-    return results.map((result) => ({
+    return results?.map((result) => ({
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticFacet/quanticFacet.js:143]: If I am not mistaken, this either returns results or `undefined` right?

If a getter that is used directly in an iterator/for:each in your templates ever return something other than an array, the whole LWC will throw and not load.

I would say, we need to make sure these functions `get facetSearchResults()` either return results or an empty array.

**Review by erocheleau — APPROVED** [review]
Make sure the getters are safe

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js [hunk 4]
```diff
@@ -118,15 +118,15 @@ export default class QuanticCategoryFacet extends LightningElement {
   }
 
   get hasSearchResults() {
-    return this.facet?.state.facetSearch.values.length !== 0;
+    return this.facet?.state.facetSearch.values?.length !== 0;
```
[the remark this answers is not in the document]
Comment (aelfaleh) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js]: @lbergeron  can you please check my last modification, Thank you

**Review by lbergeron — APPROVED** [review]
Looks good to me!

**Pull request merged**